tests/ha/test_server_pod_failure.py

```diff
@@ -0,0 +1,506 @@
+        LOGGER.info("Step 5: Services of server pod %s are in offline state", pod_name)
+
+        pod_list.remove(pod_name)
+        LOGGER.info("Step 6: Check services status on remaining pods %s", pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 6: Services on the remaining pods are in online state")
+
+        LOGGER.info("Step 7: Perform READs and verify DI on the written data")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 7: Performed READs and verified DI on the written data")
+
+        LOGGER.info("ENDED: Test to verify degraded reads after unsafe server pod shutdown.")
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-39904")
+    @CTFailOn(error_handler)
+    def test_continuous_reads_during_server_pod_down(self):
+        """Following test steps tests degraded reads while server pod is going down."""
+        LOGGER.info("STARTED: Test to verify degraded reads during server pod is going down.")
+        event = threading.Event()  # Event to be used to send intimation of server pod deletion
+
+        LOGGER.info("Step 1: Perform WRITEs with variable object sizes. 0B + (1KB - 512MB)")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix = 'test-39904'
+        self.s3_clean = users
+        output = Queue()
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    nsamples=30, nclients=20,
+                                                    log_prefix=self.test_prefix,
+                                                    skipread=True, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs with variable sizes objects.")
+
+        LOGGER.info("Step 2: Perform READs & verify DI on the written data in background")
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix,
+                'nclients': 5, 'nsamples': 30, 'skipwrite': True, 'skipcleanup': True,
+                'output': output}
+        thread = threading.Thread(target=self.ha_obj.event_s3_operation,
+                                  args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 2: Successfully started READs & verified DI on the written data in "
+                    "background")
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
```

Comment:
Please add comment for this delay.

Reply:
> Please add comment for this delay.

We are adding the same if delay goes beyong 60 sec right.

Reply:
Didn't get this?
Need to add comment for explaining why this delay is there.